kalpsdk/write_transaction.go

```diff
@@ -8,6 +8,67 @@ import (
 	res "github.com/p2eengineering/kalp-sdk-public/response"
 )
 
+// Initialize initializes the smart contract owner by setting the owner ID.
+// It checks if there's already an owner, and if not, it sets the caller's userID as the owner ID.
+//
+// Returns:
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) Initialize() error {
+	ownerId, err := ctx.GetStub().GetState("smartContractOwner")
+	if err != nil {
+		return err
+	}
+
+	if string(ownerId) != "" {
+		return nil
+	}
+
+	userId, err := ctx.GetUserID()
+	if err != nil {
+		return err
+	}
+
+	err = ctx.GetStub().PutState("smartContractOwner", []byte(userId))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TransferOwner transfers ownership of the smart contract to the caller.
```

Comment:
@chiragp2e Please update comment to indicate ownership change to new owner.

Reply:
done